### PR TITLE
[BBC-9066] Storybook: Migrate Autocomplete to MDX

### PR DESCRIPTION
**Pull request by pith**
## Description

Migrate the autocomplete component documentation to the MDX format.

## What has been done

Usage of MDX and some addition to the documentation of the autocomplete.

image

image

## Things to consider

There is no specification because the Figma specs are too long and only explain the implementation details of the autocomplete which doesn't bring value to autocomplete users.

## How it was tested

Unit test + storybook.

### CODE REVIEW

File: src/autoComplete/AutoComplete.story.mdx [hunk 1]
```diff
@@ -0,0 +1,178 @@
+import { Meta, Preview, Props, Story } from '@storybook/addon-docs/blocks'
+
+import { Button, ButtonStatus } from '../button'
+import { SearchIcon } from '../icon'
+import { BaseSection } from '../layout/section/baseSection'
+import { AutoCompleteExample } from './AutoCompleteExample'
+import { AutoComplete } from './index'
+
+<Meta title="AutoComplete" />
```
Comment (gabrielagabriel) [inline src/autoComplete/AutoComplete.story.mdx:9]:
```suggestion
<Meta title="Forms|AutoComplete" />
```
AutoComplete is a form element lets bring it to this category to improve discoverability

File: src/autoComplete/AutoComplete.story.mdx [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+
+# **AutoComplete**
+
+### Default
+
+<Preview>
+  <Story name="Default">
+    <BaseSection>
```
Comment (gabrielagabriel) [inline src/autoComplete/AutoComplete.story.mdx:17]: Please, avoid using `<BaseSection>` to stories if isn't mandatory to use it with this component

File: src/autoComplete/AutoComplete.story.mdx [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+        </Button>
+      )}
+      />
+    </BaseSection>
+  </Story>
+</Preview>
+
+## Specifications
+
+## Usage
```
Comment (gabrielagabriel) [inline src/autoComplete/AutoComplete.story.mdx:89]: Really nice usage, thanks!

File: src/autoComplete/AutoCompleteExample.tsx [hunk 4]
```diff
@@ -0,0 +1,87 @@
+import React, { useState } from 'react'
+
+import { ItemStatus } from '../_internals/item'
+import { ComfortIcon } from '../icon'
+import Loader, { LoaderLayoutMode } from '../loader'
+import { AutoComplete, AutocompleteItem, AutocompleteOnChange } from './AutoComplete'
```
Comment (gabrielagabriel) [inline src/autoComplete/AutoCompleteExample.tsx:6]:
```suggestion
import { AutoComplete, AutocompleteItem, AutocompleteOnChange } from './index'
```

File: src/autoComplete/AutoComplete.story.mdx (hunk 2 above)
Comment (pith) [reply src/autoComplete/AutoComplete.story.mdx:17]: Sadly it seems necessary

**Review by mcellier — APPROVED** [review]

**Review by wakooka — APPROVED** [review]

**Pull request merged**